crates/router/src/core/payments.rs

```diff
@@ -198,11 +195,6 @@ where
                 .await?
             }
         };
-
-        if should_delete_pm_from_locker(payment_data.payment_intent.status) {
-            vault::Vault::delete_locker_payment_method_by_lookup_key(state, &payment_data.token)
-                .await
-        }
```

Comment:
Do not delete this function call. based on the operation decide whether to delete the payment method or not

Reply:
There is an entry being made in the process tracker to delete card from vault after a specified amount of time which will delete the card details. Keeping the above function would return error when the process tracker tries to perform the delete operation as the card details would already be deleted by the above function.